core/modules/node/node.module

```diff
@@ -2912,6 +2927,40 @@ function _node_query_node_access_alter($query, $type) {
 }
 
 /**
+ * Returns the user's view_unpublished permissions in the following format:
+ *  $perms['any'] = TRUE if any of the below is TRUE, otherwise FALSE.
+ *  $perms['full'] = TRUE if "view any unpublished content" is set.
+ *  $perms['content_type'] = array of content types and the user's
+ *    associated view unpublished permission.
+*/
+function node_unpublished_user_permsissions() {
+  $permissions = FALSE;
+  $permissions['full'] = user_access('view any unpublished content') ? TRUE : FALSE;
+
+  foreach (node_permissions_get_configured_types() as $type) {
+    $type = check_plain($info->type);
+    $permission = "view any unpublished $type content";
+    $permissions['content_type'][$machine_type] = user_access($permission) ? TRUE : FALSE;
+  }
+
+  $permissions['any'] = _node_in_array_r(TRUE, $permissions, TRUE);
+
+  return $permissions;
+}
+
+/**
+ * Utility function, multidimensional in_array().
+*/
+function _node_in_array_r($needle, $haystack, $strict = TRUE) {
```

Comment:
This function seems unnecessary, as it's only used in one place. Perhaps merge with `node_unpublished_user_permsissions()`?